extensions/geode-modules/src/main/java/org/apache/geode/modules/session/catalina/ClientServerSessionCache.java

```diff
@@ -190,9 +190,8 @@ protected void createOrRetrieveRegion() {
         sessionRegion.getAttributesMutator().addCacheListener(new SessionExpirationCacheListener());
       }
 
-      // This is true for PROXY regions
-      if (sessionRegion.getAttributes().getDataPolicy() == DataPolicy.EMPTY) {
-        sessionRegion.registerInterest("ALL_KEYS", InterestResultPolicy.KEYS);
+      if (sessionRegion.getAttributes().getDataPolicy() != DataPolicy.EMPTY) {
```

Comment:
We've flipped the comparison here... is this intentional?  What happens in the EMPTY case?

Reply:
It looks like the registration is required for:
1. To handle destroyed session on client (For empty regions) and 
2. Replication (for caching-proxy region) when the application fails-over between the two gemfire clients managing the session. Is this supported/requirement?

Reply:
In either case, why did we do == EMPTY before and != EMPTY now?  Why not just do it for all cases?  Since we flipped the logic, was it completely backwards/wrong before?

Reply:
Yes, needed to register interest for both cases - still needs the callback events for empty proxy region as well.